### PR TITLE
Feature/asset tree filter improvement

### PR DESCRIPTION
**Pull request by jvervier**
Hi @richturner ,

Things move ! I'll mail you for one specific details :-)

### CODE REVIEW

**Comment by jvervier** [comment]
Hi @richturner ,

I made a proposal for the input text field issue with the absolute clear-icon which is a trailing icon with action outside the or-mwc-input.

Everything is ready to be checked

regards

File: ui/component/or-asset-tree/src/index.ts [hunk 1]
```diff
@@ -346,27 +434,102 @@ export class OrAssetTree extends subscribe(manager)(LitElement) {
                               outlined="true"
                               @input="${(e: KeyboardEvent) => {
                                   // Means some input is occurring so delay filter
-                                  this._onFilterInput(e);                                  
+                                  this._onFilterInputEven(e);
```
Comment (richturner) [inline ui/component/or-asset-tree/src/index.ts:437]: Typo

File: ui/component/or-asset-tree/src/index.ts [hunk 2]
```diff
@@ -575,30 +738,215 @@ export class OrAssetTree extends subscribe(manager)(LitElement) {
...
+    protected _onFilterInput(newValue: string | undefined, force: boolean): void {
+        if (newValue) {
+            this._clearIconContainer.classList.add("visible");
+        } else {
+            this._clearIconContainer.classList.remove("visible");
+        }
+
+        let currentFilter: OrAssetTreeFilter = this.parseFromInputFilter(newValue);
+
+        if (this._filter.asset === currentFilter.asset &&
```
Comment (richturner) [inline ui/component/or-asset-tree/src/index.ts:927]: There is a util helper method for object deepEquals checking

File: ui/component/or-asset-tree/src/index.ts (hunk 1 above)
Comment (jvervier) [reply ui/component/or-asset-tree/src/index.ts:437]: Fixed

File: ui/component/or-asset-tree/src/index.ts (hunk 2 above)
Comment (jvervier) [reply ui/component/or-asset-tree/src/index.ts:927]: You're right, I moved to Util.equals

**Comment by jvervier** [comment]
Hi @richturner , Just get your comments into the code. Pushed on the branch

File: ui/component/model/src/index.ts [hunk 3]
```diff
@@ -1,2 +1,2521 @@
-export * from "./model";
-export {AssetModelUtil} from "./util";
```
Comment (richturner) [inline ui/component/model/src/index.ts:2]: Not sure how this has happened but the autogenerated contents get output to `model.ts` but they seem to have been copied into here?

**Review by richturner — CHANGES_REQUESTED** [review]

File: ui/component/or-asset-tree/src/index.ts [hunk 4]
```diff
@@ -649,42 +998,145 @@ export class OrAssetTree extends subscribe(manager)(LitElement) {
...
+            attributes: attributeCond
         };
 
         const response = await manager.rest.api.AssetResource.queryAssets(query);
+
         const foundAssetIds: string[] = response.data.map((asset: Asset) => asset.id!);
 
         return (asset) => {
-            return foundAssetIds.includes(asset.id!);
+            let attrValueCheck = true;
```
[the remark this answers is not in the document]
Comment (jvervier) [reply ui/component/or-asset-tree/src/index.ts:1061]: My bad for that.

I'll implement the comparaison on assetQuery side

File: ui/component/model/src/index.ts (hunk 3 above)
Comment (jvervier) [reply ui/component/model/src/index.ts:2]: MMMh I do not intend myself to copy this content. Don't really know what could have create such case.

File: ui/component/or-asset-tree/src/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (jvervier) [reply ui/component/or-asset-tree/src/index.ts:1061]: @richturner The only things is : using the ValuePredicateUnion need to know the attribute type. By doing it after the call, we know the attribute data type and can handle the right comparaison.
If we need to know the Attribute type before executing the request we need to get the attribute type (linked to the attribute name provided in the search) by another way (or another request).

For that part I prefer to get your final word on this. If you prefer we can schedule a discussion about that :-)

[the remark this answers is not in the document]
Comment (jvervier) [reply ui/component/or-asset-tree/src/index.ts:1061]: Thinking about this, I could also use the content on the right of the filter attribute value "attribute:valueComparaison" and test the type of content to make assumptions we're comparing a number attribute value,... in order to use the right Type in the ValuePredicateUnion.

File: ui/component/model/src/index.ts (hunk 3 above)
Comment (jvervier) [reply ui/component/model/src/index.ts]: I don't really know which action lead to that change but I fix this by reverting the changes

**Review by richturner — APPROVED** [review]

**Pull request merged**